docs/migration_guide.md

```diff
@@ -0,0 +1,105 @@
+# Migrating from bincode 1 to 2
+
+Bincode 2 now has an optional dependency on `serde`. You can either use `serde`, or use bincode's own `derive` feature and macros.
+
+## From `Options` to `Configuration`
+
+Bincode 1 had the [Options](https://docs.rs/bincode/1/bincode/config/trait.Options.html) trait. This has been replaced with the [Configuration](https://docs.rs/bincode/2.0.0-beta/bincode/config/struct.Configuration.html) struct.
+
+If you're using `Options`, you can change it like this:
+
+```rust,ignore
+# old
+bincode_1::DefaultOptions::new().with_fixint_encoding()
+
+# new
+bincode_2::config::legacy().with_fix_int_encoding()
+```
+
+Make sure to use `config::legacy()` if you want to be binary compatible with bincode 1. If you do not care about this, we recommend using the `config::standard()`.
+
+The following changes have been made:
+- `.with_limit(n)` has been changed to `.with_limit::<n>()`.
+- `.with_native_endian()` has been removed. Use `.with_big_endian()` or `with_little_endian()` instead.
+- `.with_varint_encoding()` has been renamed to `.with_variable_int_encoding()`.
+- `.with_fixint_encoding()` has been renamed to `.with_fixed_int_encoding()`.
+- `.reject_trailing_bytes()` has been removed.
+- `.allow_trailing_bytes()` has been removed.
+- You can no longer (de)serialize from the `Options` trait directly. Use one of the `encode_` or `decode_` methods.
+
+Because of confusion with `Options` defaults in bincode 1, we have made `Configuration` mandatory in all calls in bincode 2.
+
+## Migrating with `serde`
+
+Make sure to include bincode 2 with the `serde` feature enabled.
+
+```toml
+[dependencies]
+bincode = { version = "2.0.0-beta", features = ["serde"] }
+
+# Optionally you can disable the `derive` feature:
+# bincode = { version = "2.0.0-beta", default-features = false, features = ["std", "serde"] }
+```
+
+Then replace the following functions:
+
+|Bincode 1|Bincode 2|
```

Comment:
Specify the `Configuration` that should be used in this table to get the equivalent function call